crates/re_viewport/src/context_menu/actions.rs

```diff
@@ -89,6 +103,66 @@ impl ContextMenuAction for HideAction {
             false,
         );
     }
+
+    fn process_data_result(
+        &self,
+        ctx: &ContextMenuContext<'_>,
+        space_view_id: &SpaceViewId,
+        instance_path: &InstancePath,
+    ) {
+        set_data_result_visible(ctx, space_view_id, instance_path, false);
+    }
+}
+
+fn data_result_visible(
+    ctx: &ContextMenuContext<'_>,
+    space_view_id: &SpaceViewId,
+    instance_path: &InstancePath,
+) -> Option<bool> {
+    if instance_path.is_splat() {
+        if let Some(space_view) = ctx.viewport_blueprint.space_view(space_view_id) {
+            let query_result = ctx
+                .viewer_context
+                .lookup_query_result(space_view.query_id());
+            query_result
+                .tree
+                .lookup_result_by_path(&instance_path.entity_path)
+                .map(|data_result| {
+                    data_result
+                        .recursive_properties()
+                        .map_or(true, |prop| prop.visible)
+                })
+        } else {
+            None
+        }
+    } else {
+        None
+    }
```

Comment:
I personally find the functional approach much easier to read here but YMMV.

```suggestion
    instance_path
        .is_splat()
        .then(|| {
            ctx.viewport_blueprint
                .space_view(space_view_id)
                .and_then(|space_view| {
                    let query_result = ctx
                        .viewer_context
                        .lookup_query_result(space_view.query_id());
                    query_result
                        .tree
                        .lookup_result_by_path(&instance_path.entity_path)
                        .map(|data_result| {
                            data_result
                                .recursive_properties()
                                .map_or(true, |prop| prop.visible)
                        })
                })
        })
        .flatten()
```